networkit/cpp/io/test/IOGTest.cpp

```diff
@@ -1225,4 +1225,18 @@ TEST_F(IOGTest, testNetworkitBinaryZigzag) {
     }
 }
 
+TEST_F(IOGTest, testNetworkitWriterNonContinuousNodesIds) {
+    Graph G(20, true);
+    G.removeNode(10);
+    NetworkitBinaryReader reader;
+    NetworkitBinaryWriter writer;
+    std::string path = "output/test.gt";
+    writer.write(G, path);
+    Graph Gread = reader.read(path);
+    EXPECT_EQ(G.numberOfNodes(), Gread.numberOfNodes());
+    EXPECT_EQ(G.numberOfEdges(), Gread.numberOfEdges());
+    EXPECT_EQ(G.isDirected(), Gread.isDirected());
+    EXPECT_EQ(G.isWeighted(), Gread.isWeighted());
```

Comment:
This only checks the **number** of nodes and edges. Shouldn't we also check that the two graphs are the same? My suggestion is to add such a check here for now. Then, in another PR, we could implement it in the `==` operator in `Graph` or as a utility function in `GraphTools`.

Reply:
Maybe for this function it is good to have both. Implment it in `GraphTools`, but also overload the operator and call the `GraphTools` function.

Reply:
For two graphs to be the same (not isomorphism) we also need to check the existence of every node/edge (+ possible weights). For edges also the endpoints should be the same.